src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySink.java

```diff
@@ -218,6 +220,13 @@ void recordMetric(boolean succeeded, BatchSinkContext context) {
       context.getMetrics().count(RECORDS_UPDATED_METRIC, rowCount);
       totalRows -= rowCount;
     }
+
+    Map<String, String> tags = new ImmutableMap.Builder<String, String>()
+        .put(Constants.Metrics.Tag.APP_ENTITY_TYPE, BatchSink.PLUGIN_TYPE)
```

Comment:
this seems like something that should be done at the application level instead of forcing the plugins to know about these special tags and potentially putting the wrong values. For example, maybe just make the existing context.getMetrics() call add these tags? Can you open an improvement jira?

Reply:
sg, done: https://cdap.atlassian.net/browse/PLUGIN-1626